omero/developers/Web/LinkingFromWebclient.txt

```diff
@@ -0,0 +1,189 @@
+For example, if you have webtest installed you can use the ``channel_overlay_viewer``:
+
+::
+
+    $ bin/omero config set omero.web.viewer.view webtest.views.channel_overlay_viewer
+
+This will now direct the image viewer url at ``webclient/img_detail/<iid>/`` to this viewer.
+However, the existing viewer will still be available under webgateway at ``webgateway/img_detail/<iid>/``.
+
+If you want to use a different viewer for different images, you can conditionally redirect to
+the webgateway viewer or elsewhere.
+For example:
+
+::
+
+    if image.getSizeC() == 1:
+        return HttpResponseRedirect(reverse("webgateway.views.full_viewer", args=(iid,)))
+
+
+Open with
+---------
+
+The 'Open with' configuration allows users to 'open' data from OMERO in another web page
+using :property:`omero.web.open_with`.
+
+For example:
+
+ - Open images in a custom viewer
+ - Add images to OMERO.figure
+ - Link to external resources, E.g. open Dataset named '000397' with url https://www.ncbi.nlm.nih.gov/protein/000397
+
+Currently, 'Open With' options are shown in the context menu of the left-panel tree
+and are therefore only available for objects shown in the tree.
+
+Open image in viewer
+^^^^^^^^^^^^^^^^^^^^
+
+In the simplest case the minimum needed to add a viewer is "label" and "url_name".
+
+::
+
+    $ bin/omero config append omero.web.open_with '["My viewer", "url_name"]'
+
+We use ``reverse(url_name)`` to resolve a url from the url_name. If the url_name
+is not recognised (E.g. for external urls) the ``url_name`` will be used directly.
+
+By default, the ``Open With > My viewer`` menu option will only be enabled when a
+single image is selected. When the user clicks the image, a new popup window will be opened
+with the specified url, passing the image ID in the query string:
+
+::
+
+    url?image=:id
+
+Further parameters can be specified in an options object, as described below.
+
+Open in new tab
+^^^^^^^^^^^^^^^
+
+If you wish to open in a new browser tab instead of a popup window, you can add a ``target``
+attribute to the options:
+
+::
+
+    $ bin/omero config append omero.web.open_with '["My viewer", "url_name"], {"target":"_blank"}]'
+
+Supported objects
+^^^^^^^^^^^^^^^^^
+
+By default, 'Open with' options are enabled when only a single image is selected.
+This can be configured to support multiple images or other data types by adding a
+``supported_objects`` list to the options.
+Specify data types as E.g. ``project, dataset, image, screen, plate, acquisition``
+or use plurals to indicate that multiple objects are supported.
+E.g. ```images``` will enable the 'Open with' option when 1 or more images are selected.
+In the following example, we support a single ``dataset`` or multiple ``images``.
+
+::
+
+    $ bin/omero config append omero.web.open_with '["My viewer", "url_name"], {"supported_objects":["dataset", "images"]}]'
```

Comment:
As this is a single line example, it would be neater to put the `::` after `images` so you don't need all these extra lines, just one blank line and then the example.